cratedb_toolkit/io/mongodb/api.py

```diff
@@ -135,32 +139,26 @@ def mongodb_copy(
                 progress=progress,
             )
         )
+
+    # `full-load` procedure, multiple collections.
     else:
         logger.info(f"Inquiring collections at {source_url}")
-        mongodb_uri = source_url
-        cratedb_uri = target_url
-        # What the hack?
-        if (
-            mongodb_uri.scheme.startswith("mongodb")
-            and Path(mongodb_uri.path).is_absolute()
-            and mongodb_uri.path[-1] != "/"
-        ):
-            mongodb_uri.path += "/"
-        if cratedb_uri.path[-1] != "/":
-            cratedb_uri.path += "/"
-        mongodb_query_parameters = mongodb_uri.query_params
-        mongodb_adapter = mongodb_adapter_factory(mongodb_uri)
+        address_pair_root = AddressPair(source_url=source_url, target_url=target_url)
```

Comment:
This hackery has now also been refactored and wrapped away into an `AddressPair` object, in order to separate concerns...